npm-scripts.js

```diff
@@ -14,140 +16,138 @@ const MAYOR_VERSION = version.split('.')[0];
+			{
+				cleanWorker();
+			}
 		}
 
-		execute('tsc --project node');
-		taskReplaceVersion();
+		break;
+	}
+
+	case 'typescript:build':
+	{
+		buildTypescript(/* force */ true);
+		replaceVersion();
 
 		break;
 	}
 
 	case 'typescript:watch':
 	{
-		const TscWatchClient = require('tsc-watch/client');
-
-		if (!isWindows)
-		{
-			execute('rm -rf node/lib');
-		}
-		else
-		{
-			execute('rmdir /s /q "node/lib"');
-		}
+		deleteNodeLib();
 
+		const TscWatchClient = require('tsc-watch/client');
 		const watch = new TscWatchClient();
 
-		watch.on('success', taskReplaceVersion);
+		watch.on('success', replaceVersion);
 		watch.start('--project', 'node', '--pretty');
 
 		break;
 	}
 
 	case 'worker:build':
 	{
-		execute(`${MAKE} -C worker`);
+		buildWorker();
 
 		break;
 	}
 
 	case 'lint:node':
 	{
-		execute('eslint -c node/.eslintrc.js --max-warnings 0 node/src/ node/.eslintrc.js npm-scripts.js node/tests/ worker/scripts/gulpfile.js');
+		lintNode();
 
 		break;
 	}
 
 	case 'lint:worker':
 	{
-		execute(`${MAKE} lint -C worker`);
+		lintWorker();
 
 		break;
 	}
 
 	case 'format:worker':
 	{
-		execute(`${MAKE} format -C worker`);
+		executeCmd(`${MAKE} format -C worker`);
 
 		break;
 	}
 
 	case 'test:node':
 	{
-		taskReplaceVersion();
-
-		if (!process.env.TEST_FILE)
-		{
-			execute('jest');
-		}
-		else
-		{
-			execute(`jest --testPathPattern ${process.env.TEST_FILE}`);
-		}
+		buildTypescript(/* force */ false);
+		replaceVersion();
+		testNode();
 
 		break;
 	}
 
 	case 'test:worker':
 	{
-		execute(`${MAKE} test -C worker`);
-		break;
-	}
-
-	case 'coverage':
-	{
-		taskReplaceVersion();
-		execute('jest --coverage');
-		execute('open-cli coverage/lcov-report/index.html');
+		testWorker();
 
 		break;
 	}
 
-	case 'postinstall':
+	case 'coverage':
 	{
-		if (!process.env.MEDIASOUP_WORKER_BIN)
-		{
-			execute('node npm-scripts.js worker:build');
-			// Clean build artifacts except `mediasoup-worker`.
-			execute(`${MAKE} clean-build -C worker`);
-			// Clean downloaded dependencies.
-			execute(`${MAKE} clean-subprojects -C worker`);
-			// Clean PIP/Meson/Ninja.
-			execute(`${MAKE} clean-pip -C worker`);
-		}
+		buildTypescript(/* force */ false);
+		replaceVersion();
+		executeCmd('jest --coverage');
+		executeCmd('open-cli coverage/lcov-report/index.html');
 
 		break;
 	}
 
 	case 'release':
 	{
-		execute('node npm-scripts.js typescript:build');
-		execute('npm run lint');
-		execute('npm run test');
-		execute(`git commit -am '${version}'`);
-		execute(`git tag -a ${version} -m '${version}'`);
-		execute(`git push origin v${MAYOR_VERSION} && git push origin --tags`);
-		execute('npm publish');
+		buildTypescript(/* force */ true);
+		buildWorker();
+		lintNode();
+		lintWorker();
+		testNode();
+		testWorker();
+		executeCmd(`git commit -am '${version}'`);
+		executeCmd(`git tag -a ${version} -m '${version}'`);
+		executeCmd(`git push origin v${MAYOR_VERSION} && git push origin --tags`);
```

Comment:
What about pushing only single tag you've just created instead?

Reply:
I assume workflow here is:

- We are in v3 branch.
- Repo is clean.
- Change version in `package.json`.
  - NOTE: Missing thing here, command to update mediasoup version in `package-lock.json`, I'll add it.
- Run `npm run release`.

So `git push origin` is indeed needed.

Reply:
I was commenting about `--tags`, you don't need to push *all* tags, just the new tag created here.

BTW, there is no check that you are on `v3` branch here.

Reply:
I don't know how to push only a tag.

> BTW, there is no check that you are on v3 branch here.

Can we please improve this in a separate effort? This was already here for long.

Reply:
Quite simple:
```javascript
executeCmd(`git push origin '${version}'`);
```

Reply:
Do you mean replacing

```
executeCmd(`git push origin v${MAYOR_VERSION} && git push origin --tags`);
```

with:

```
executeCmd(`git push origin v${MAYOR_VERSION} && git push origin '${version}'`);
```

?

Reply:
Done in https://github.com/versatica/mediasoup/pull/954/commits/a962fdea7bdb13101638aa83aea7e036b2fda454